java-engine/src/main/java/io/getunleash/engine/UnleashEngine.java

```diff
@@ -69,26 +87,41 @@ public static byte[] buildMessage(String toggleName, Context context) {
         context.getEnvironment() != null ? builder.createString(context.getEnvironment()) : 0;
 
     List<Map.Entry<String, String>> entries = new ArrayList<>(context.properties.entrySet());
-    int[] propertyOffsets = new int[entries.size()];
-    for (int i = 0; i < entries.size(); i++) {
-      Map.Entry<String, String> entry = entries.get(i);
+    List<Integer> offsets = new ArrayList<>();
+    for (Map.Entry<String, String> entry : entries) {
```

Comment:
Small adjustment to the context properties code here - basically strip out keys with null values. We have a spec test that passes this context:
```
            "context": {
                "properties": {
                    "customFieldMissing": null
                }
            }
```